crates/extensions/c8y_mapper_ext/src/service_monitor.rs

```diff
@@ -23,7 +24,7 @@ pub fn is_c8y_bridge_established(
 pub fn convert_health_status_message(
     mqtt_schema: &MqttSchema,
     entity: &EntityMetadata,
-    ancestors_external_ids: &[String],
+    parent_external_id: Option<&EntityExternalId>,
```

Comment:
Why is the id of the parent now enough while the whole ancestor list was previously required?

Reply:
Maintaining the entire ancestors list was not really required to create or publish the data to a nested entity. We didn't know that earlier (when child device support was added). We later realised that, once a child device is created at any level with an external id (XID), its nested child device can be created by simply publishing to `s/us/<xid>` directly without the entire ancestry in the topic. This is possible because XIDs are unique for a tenant and hence there can't be 2 different entities with the same XID at different levels in the entity hierarchy.

With the introduction of the entity cache, I just used this opportunity to clean that up, as that helped me simplify and minimize the metadata maintained by the entity cache.